CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -438,6 +487,85 @@ void SiStripGainFromCalibTree::bookDQMHistos(const char* dqm_dir, const char* ta
     Charge_Vs_PathlengthTECP2[elepos] = dbe->book2S(cvpTECP2.c_str(), cvpTECP2.c_str(), 20   , 0.3 , 1.3  , 250,0,2000);
     Charge_Vs_PathlengthTECM1[elepos] = dbe->book2S(cvpTECM1.c_str(), cvpTECM1.c_str(), 20   , 0.3 , 1.3  , 250,0,2000);
     Charge_Vs_PathlengthTECM2[elepos] = dbe->book2S(cvpTECM2.c_str(), cvpTECM2.c_str(), 20   , 0.3 , 1.3  , 250,0,2000);
+
+    //Book Charge monitoring histograms
+    std::vector<std::pair<std::string,std::string>> hnames = APVGain::monHnames(VChargeHisto,doChargeMonitorPerPlane,"");
+    for (unsigned int i=0;i<hnames.size();i++){
+        std::string htag = (hnames[i]).first + stag;
+        Charge_1[elepos].push_back( dbe->book1DD( htag.c_str(), (hnames[i]).second.c_str(), 100   , 0. , 1000. ) );
+    }
+
+    hnames = APVGain::monHnames(VChargeHisto,doChargeMonitorPerPlane,"woG2");
+    for (unsigned int i=0;i<hnames.size();i++){
+        std::string htag = (hnames[i]).first + stag;
+        Charge_2[elepos].push_back( dbe->book1DD( htag.c_str(), (hnames[i]).second.c_str(), 100   , 0. , 1000. ) );
+    }
+
+    hnames = APVGain::monHnames(VChargeHisto,doChargeMonitorPerPlane,"woG1");
+    for (unsigned int i=0;i<hnames.size();i++){
+        std::string htag = (hnames[i]).first + stag;
+        Charge_3[elepos].push_back( dbe->book1DD( htag.c_str(), (hnames[i]).second.c_str(), 100   , 0. , 1000. ) );
+    }
+
+    hnames = APVGain::monHnames(VChargeHisto,doChargeMonitorPerPlane,"woG1G2");
+    for (unsigned int i=0;i<hnames.size();i++){
+        std::string htag = (hnames[i]).first + stag;
+        Charge_4[elepos].push_back( dbe->book1DD( htag.c_str(), (hnames[i]).second.c_str(), 100   , 0. , 1000. ) );
+    }
+
+    //Book validation histograms
+    if (m_harvestingMode) {
+        MPV_Vs_EtaTIB  = dbe->book2DD("MPV_Vs_EtaTIB" ,"MPV_Vs_EtaTIB" , 50, -3.0, 3.0, 300, 0, 600);
```

Comment:
define 300, 0, 600 (at least) as local variables to avoid repeating them many times